File: backend/cp25ssi2/src/main/java/sit/chat2date/cp25ssi2/dto/PreferenceDto.java
package sit.chat2date.cp25ssi2.dto;

import lombok.Builder;
import lombok.Data;
import sit.chat2date.cp25ssi2.entities.Interest;
import sit.chat2date.cp25ssi2.entities.LifeStyle;
import sit.chat2date.cp25ssi2.entities.Tag;
import sit.chat2date.cp25ssi2.entities.TravelStyle;

import java.util.List;

@Builder
@Data
public class PreferenceDto {
    private List<Interest> interests;
    private List<LifeStyle> lifeStyles;
    private List<Tag> tags;
    private List<TravelStyle> travelStyles;
}

File: frontend-mobile/chat2date/lib/models/dto/preference_dto.dart
import 'package:chat2date/models/interest.dart';
import 'package:chat2date/models/lifestyle.dart';
import 'package:chat2date/models/tag.dart';
import 'package:chat2date/models/travelstyle.dart';

class PreferenceDto {
  final List<Travelstyle> travelStyles;
  final List<Lifestyle> lifeStyles;
  final List<Interest> interests;
  final List<Tag> tags;

  PreferenceDto({
    required this.travelStyles,
    required this.lifeStyles,
    required this.interests,
    required this.tags
  });

  factory PreferenceDto.fromJson(Map<String, dynamic> json) {
    return PreferenceDto(
      tags: (json['tags'] as List).map((x) => Tag.fromJson(x)).toList(),
      travelStyles:
          (json['travelStyles'] as List).map((x) => Travelstyle.fromJson(x)).toList(),
      lifeStyles:
          (json['lifeStyles'] as List).map((x) => Lifestyle.fromJson(x)).toList(),
      interests:
          (json['interests'] as List).map((x) => Interest.fromJson(x)).toList(),
    );
  }
}